File: src/main/java/org/pixeltime/enchantmentsenhance/event/enchantment/Stealth.java
package org.pixeltime.enchantmentsenhance.event.enchantment;

import com.sk89q.worldguard.bukkit.WGBukkit;
import com.sk89q.worldguard.protection.flags.DefaultFlag;
import com.sk89q.worldguard.protection.flags.StateFlag;
import org.bukkit.ChatColor;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerToggleSneakEvent;
import org.bukkit.inventory.ItemStack;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

public class Stealth implements Listener {
    @EventHandler
    public void onSneak(final PlayerToggleSneakEvent playerToggleSneakEvent) {
        final Player player = playerToggleSneakEvent.getPlayer();
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantment." + "stealth"));
        if (SettingsManager.enchant.getBoolean("allow-worldguard") && WGBukkit.getRegionManager(player.getWorld()).getApplicableRegions(player.getLocation()).queryState(null, new StateFlag[]{DefaultFlag.PVP}) == StateFlag.State.DENY) {
            return;
        }
        final int n = (int) (Math.random() * 100.0);
        final ItemStack[] armorContents = player.getInventory().getArmorContents();
        try {
            ItemStack[] array;
            for (int length = (array = armorContents).length, i = 0; i < length; ++i) {
                final ItemStack itemStack = array[i];
                if (itemStack != null) {
                    if (itemStack.hasItemMeta() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I") && n < SettingsManager.enchant.getInt("stealth.level_I.chance")) {
                        final int int1 = SettingsManager.enchant.getInt("stealth.level_I.radius");
                        for (final Entity entity : player.getNearbyEntities((double) int1, (double) int1, (double) int1)) {
                            if (entity instanceof Player) {
                                ((Player) entity).addPotionEffect(new PotionEffect(PotionEffectType.BLINDNESS, SettingsManager.enchant.getInt("stealth.level_I.duration") * 20, 0));
                            }
                        }
                    }
                    if (itemStack.hasItemMeta() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " II") && n < SettingsManager.enchant.getInt("stealth.level_II.chance")) {
                        final int int2 = SettingsManager.enchant.getInt("stealth.level_II.radius");
                        for (final Entity entity2 : player.getNearbyEntities((double) int2, (double) int2, (double) int2)) {
                            if (entity2 instanceof Player) {
                                ((Player) entity2).addPotionEffect(new PotionEffect(PotionEffectType.BLINDNESS, SettingsManager.enchant.getInt("stealth.level_II.duration") * 20, 0));
                            }
                        }
                    }
                    if (itemStack.hasItemMeta() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " III") && n < SettingsManager.enchant.getInt("stealth.level_III.chance")) {
                        final int int3 = SettingsManager.enchant.getInt("stealth.level_III.radius");
                        for (final Entity entity3 : player.getNearbyEntities((double) int3, (double) int3, (double) int3)) {
                            if (entity3 instanceof Player) {
                                ((Player) entity3).addPotionEffect(new PotionEffect(PotionEffectType.BLINDNESS, SettingsManager.enchant.getInt("stealth.level_III.duration") * 20, 0));
                            }
                        }
                    }
                    if (itemStack.hasItemMeta() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " IV") && n < SettingsManager.enchant.getInt("stealth.level_IV.chance")) {
                        final int int4 = SettingsManager.enchant.getInt("stealth.level_IV.radius");
                        for (final Entity entity4 : player.getNearbyEntities((double) int4, (double) int4, (double) int4)) {
                            if (entity4 instanceof Player) {
                                ((Player) entity4).addPotionEffect(new PotionEffect(PotionEffectType.BLINDNESS, SettingsManager.enchant.getInt("stealth.level_IV.duration") * 20, 0));
                            }
                        }
                    }
                }
            }
        } catch (Exception ex) {
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantment/Stealth.kt
/*
 *     Copyright (C) 2017-Present HealPotion
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */

package org.pixeltime.enchantmentsenhance.event.enchantment

import com.sk89q.worldguard.bukkit.WGBukkit
import com.sk89q.worldguard.protection.flags.DefaultFlag
import com.sk89q.worldguard.protection.flags.StateFlag
import org.bukkit.ChatColor
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerToggleSneakEvent
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType
import org.pixeltime.enchantmentsenhance.manager.IM
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

class Stealth : Listener {
    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
    fun onSneak(playerToggleSneakEvent: PlayerToggleSneakEvent) {
        val player = playerToggleSneakEvent.player
        val translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "stealth"))
        if (SettingsManager.enchant.getBoolean("allow-worldguard") && WGBukkit.getRegionManager(player.world).getApplicableRegions(player.location).queryState(null, *arrayOf(DefaultFlag.PVP)) == StateFlag.State.DENY) {
            return
        }

        try {
            val level = IM.getHighestLevel(player, translateAlternateColorCodes)
            if (level > 0 && (Math.random() * 100.0).toInt() < SettingsManager.enchant.getInt("stealth.$level.chance")) {
                val int1 = SettingsManager.enchant.getInt("stealth.$level.radius")
                for (entity in player.getNearbyEntities(int1.toDouble(), int1.toDouble(), int1.toDouble())) {
                    if (entity is Player) {
                        entity.addPotionEffect(PotionEffect(PotionEffectType.BLINDNESS, SettingsManager.enchant.getInt("stealth.$level.duration") * 20, 0))
                    }
                }
            }
        } catch (ex: Exception) {
        }
    }
}
